dataframe.ts

```diff
@@ -0,0 +1,242 @@
+    df.columns.addNewBool('newColumn');
+    expect(typeof(df.get('newColumn', 1)), 'boolean');
+  });
+
+  test('column list addNewDateTime', async () => {
+    return df.columns.addNewDateTime('newColumn');
+  });
+  
+  test('column list addNewFloat', async () => {
+    df.columns.addNewFloat('newColumn');
+    expect(typeof(df.get('newColumn', 1)), 'number');
+  });
+
+  test('column list addNewInt', async () => {
+    df.columns.addNewInt('newColumn');
+    expect(typeof(df.get('newColumn', 1)), 'number');
+  });
+
+  test('column list addNewQnum', async () => {
+    df.columns.addNewQnum('newColumn');
+    expect(typeof(df.get('newColumn', 1)), 'number');
+  });
+
+  test('column list addNewString', async () => {
+    df.columns.addNewString('newColumn');
+    expect(typeof(df.get('newColumn', 1)), 'string');
+  });
+
+  test('column list byIndex', async () => {
+    expect(df.columns.byIndex(1).getRawData().toString(), '1,4,2,3');
+  });
+
+  test('column list byName', async () => {
+    expect(df.columns.byName('population').getRawData().toString(), '1,4,2,3');
+  });
+
+  test('column list bySemType', async () => {
+    expect(df.columns.bySemType(null).toString(), 'countries');
+  });
+
+  test('column list bySemTypeAll', async () => {
+    expect(df.columns.bySemTypeAll(null).toString(), 'countries,population');
+  });
+
+  test('column list contains', async () => {
+    expect(df.columns.contains('population'), true);
+  });
+
+  test('column list getUnusedName', async () => {
+    expect(df.columns.getUnusedName('population').toString(), 'population (2)')
+  });
+
+  test('column list insert', async () => {
+    let newColumn = DG.Column.fromStrings('data', ['12', '34']);
+    df1.columns.insert(newColumn);
+    expect(df1.get('data', 1).toString(), '34');
+  });
+
+  test('column list names', async () => {
+    expect(df.columns.names().toString(), 'countries,population');
+  });
+
+  test('column list remove', async () => {
+    df.columns.remove('population');
+    expect(df.columns.names().toString(), 'countries');
+  });
+
+  test('column list replace', async () => {
+    let newColumn = DG.Column.fromStrings('data', ['12', '34']);
+    df1.columns.replace('countries', newColumn);
+    expect(df1.columns.names().toString(), 'data,population');
+  });
+
+  test('column list toList', async () => {
+    return df1.columns.toList();
+  });
+
+  test('column list toString', async () => {
+    return df1.columns.toString();
+  });
+
+  test('row list addNew', async () => {
+    df1.rows.addNew(['12', '23']);
+    expect(df1.get('countries', 2).toString(), '12');
+  });
+
+  test('row list filter', async () => {
+    return df1.rows.filter((row) => row.countries === 'USA');
+  });
+
+  test('row list insertAt', async () => {
+    df1.rows.insertAt(2,2);
+    expect(df1.get('countries', 2), '');
+  });
+
+  test('row list match', async () => {
+    return df1.rows.match('countries = USA').highlight();
```

Comment:
Would be nice to check if matching returns proper answers (for instance, you can check the number of matching rows)